### PR TITLE
feat(react): add useExecuteQuote hook

### PR DESCRIPTION
**Pull request by Trynax**
## Changes

Implements `useExecuteQuote` in `@spandex/react` as the React/Wagmi mutation hook for executing a selected quote.

This PR:

- adds `useExecuteQuote` to wrap core `executeQuote`
- defaults `swap.chainId` and `swapperAccount` from Wagmi connection state, like the existing quote hooks
- supports hook-level and per-call `allowanceMode`
- exports the hook from `@spandex/react`
- adds focused React tests for execution flow and override behavior
- renames the docs placeholder from `useQuoteExecutor` to `useExecuteQuote`
- updates the React hooks docs nav entry
- includes a changeset for the new public React API

## Checklist

- [x] I have followed the steps in the [Contributing guide](https://github.com/withfabricxyz/spandex/blob/main/CONTRIBUTING.md).
- [ ] I have tested this code locally with `bun test`.

## Release Impact

- [x] This change impacts released packages, and I have created a [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md).
- [ ] This change does not impact released packages (tests,docs,examples) (no release).

### CODE REVIEW

File: packages/react/lib/hooks/useExecuteQuote.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+      if (!walletClient) {
+        throw new Error("No WalletClient available from wagmi. Connect a wallet before executing.");
+      }
+
+      const publicClient = wagmiConfig
+        .getClient({ chainId: resolvedSwap.chainId })
+        ?.extend(publicActions) as PublicClient | undefined;
+
+      return coreExecuteQuote({
```
Comment (dansimpson) [inline packages/react/lib/hooks/useExecuteQuote.ts]: So, core doesn't really assume a human in the loop, so it can sequence approval then execute. With UIs, we have to account for various UX issues. For instance, on iOS, in order for the browser to switch to an app, like metamask (walletconnect style), the tap needs to happen within a second of the app open request. So, a sequenced approve, then swap will create issues that may take the user to the app store, etc. It stands to reason that buildCalls plus an intentional multi-click workflow would work better in that scenario. Obviously if the user has a smart wallet, batching will take over. 

This is better than what we have (nothing), but given some of my scar tissue around wallet connect and multi-step flows, it might make sense to abstract that away and handle the various cases for devs.

If this works for you, I am happy to approve. Curious about your thoughts on this (maybe the rules have changed on iOS, it has been awhile since I bled there)

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Thanks for the review @dansimpson 

I looked into this more and I think your concern is real.

From what I found, iOS/mobile wallet flows still seem sensitive around deep links, redirects, and timing after the initial connection, especially for WalletConnect-style sign/tx requests. So I don’t think this should present a sequenced `approve -> swap` browser flow as universally reliable.

I think the better shape is:

- auto-execute when batching is available
- auto-execute when the route only needs one call
- treat multi-call non-batched execution as a separate ui case instead of hiding it behind one mutation

That would let `useExecuteQuote` stay useful without overpromising on mobile EOA flows.

If that direction sounds right, I can update this pr so the hook builds calls first, checks the execution path, and surfaces approval-plus-swap as an explicit multi-step case when batching is not available.

Comment (dansimpson) [reply packages/react/lib/hooks/useExecuteQuote.ts]: @Trynax Thanks. I think the branching you describe will result in better outcomes. It abstracts complexity in 2 cases and exposes it in the two step flow. Seems like the right trade off.

Consider the UI dev and how they will need to surface what is happening to the user. Eg: They should have the data to know if they are doing a single call, or need to step through calls, and when they do step through those calls, there should be state/data which allows them to convey what the user is doing.. eg: The button can say approve, and there can be a sub-text item saying "Step 1 of 2" etc.

This is a fairly tricky problem to get right, and we can ship something that gets it 90% there.  Appreciate it.

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: I’ve updated the hook in that direction.

It now builds calls first and branches into:
- single-call execution
- batched execution when atomic calls are supported
- stepped execution for multi-call non-batched flows

For the stepped case, the hook now exposes UI state so apps can show the current action and step progress, like `Approve` and `Step 1 of 2`, instead of hiding approval + swap behind one mutation.

tightened a couple of edge cases around plan preparation and `allowanceMode` overrides, and added tests.

@dansimpson

Comment (dansimpson) [reply packages/react/lib/hooks/useExecuteQuote.ts]: @Trynax Nice, thanks. @dougalcantara can you give this a DX test with the example and see how it fits in?

Comment (dougalcantara) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Thanks @Trynax - two notes on the approval/execution flow after plugging it in to the example:

1) in a stepped approve->swap, the user's progress would get wiped out if a new quote comes in between the two clicks (our example refetches every 10s, so this was noticeable). `progressKey` includes `quote.txData.data`, so an incoming quote changes it, and the `useEffect` resets the step state. I'd like to propose a solution: the hook freezes the quote while `isExecuting`, and releases the freeze on error or onComplete.
2) We can lean on `useCapabilities` instead of calling `getCapabilities` - it's cached/reactive and solves an edge case around chain resolution. Also - MetaMask wasn't being detected as EIP-5792 capable, even though it is. You should expand the narrow `atomic.status === "supported"` check, MM reports "ready", for example.

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Thanks for the feedback @dansimpson @dougalcantara, I’ve updated the hook in that direction.

It now freezes the active quote during stepped execution so quote refetches do not wipe `approve -> swap` progress between clicks, and batching now uses Wagmi `useCapabilities` with both `atomic: "supported"` and `atomic: "ready"` treated as batch-capable.

Comment (dougalcantara) [reply packages/react/lib/hooks/useExecuteQuote.ts]: @Trynax looks great. One more note from me: the hook internals here are good. UIs will probably want to do/show something while the execution is happening (loading spinner, etc) . A suggestion: add an `isExecuting` to the hook's return value so that the UI can react to in-flight execution:

```tsx
const isExecuting = result.isPending || !!frozenQuote;

return {
  ...result,
  // ...
  isExecuting,
}; 
```

With this, I think we're in a really good spot with the new hook.

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Thanks @dougalcantara, added `isExecuting` so the UI has a clearer execution-state signal.

Also, while working through this flow I wrote up a follow-up idea here for the planning side of the dx: #134

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: bumping this @dansimpson @dougalcantara

Comment (dougalcantara) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Hey @Trynax thanks for the bump. This looks good to me. We'll have Dan take a look and see if he has any merge blockers.

Comment (Trynax) [reply packages/react/lib/hooks/useExecuteQuote.ts]: Thanks for the feedback @dougalcantara

Also happy to hear any thoughts on the follow-up discussion in #134.

**Review by dougalcantara — APPROVED** [review]
